code/datums/martial.dm

```diff
@@ -554,3 +554,32 @@
 		return 1
 	else
 		return 0
+
+/obj/item/clothing/suit/sleeping_carp
+	name = "carphide vest"
+	desc = "A strange armor vest formed of toughened space carp hide. Excellent at deflecting energy blasts. <span class='warning'>The collar is lined with a circlet of razor-sharp teeth.</span>"
+	icon_state = "carpvest"
+	item_state = "armor"
+	armor = list(melee = 60, bullet = 20, laser = 100, energy = 100, bomb = 5, bio = 0, rad = 0) //Complete protection from energy blasts
```

Comment:
min(armor,90) means that giving an armor 100% armoring isn't practical. You could always cheat and give it deflection mechanics though.